File: UnFinished/LeetCode_890_Find and Replace Pattern.go
package main

import "fmt"

func findAndReplacePattern(words []string, pattern string) []string {
	var res []string

	for _, word := range words {
		//ruleKeys := reflect.ValueOf(rule).MapKeys()
		rule := make(map[string]string)
		check := true
		for key, value := range word{
			if len(rule[string(pattern[key])]) == 0{
				rule[string(pattern[key])] = string(value)
			}else if rule[string(pattern[key])] != string(value){
				check = false
				break
			}
		if check{
			res = append(res, word)
		}}
	}
	return res
}

func main()  {
	words := []string{"abc","deq","mee","aqq","dkd","ccc"}
	pattern := "abb"
	result := findAndReplacePattern(words, pattern)
	fmt.Print(result)
}

File: Code/LeetCode_890_Find and Replace Pattern.py
class Solution:
    def findAndReplacePattern(self, words: 'List[str]', pattern: 'str') -> 'List[str]':
        rule = {}
        res = []

        for word in words:
            check = True
            for index, alpha in enumerate(word):
                if (pattern[index] not in rule.keys()) and (alpha not in rule.values()):
                    rule[pattern[index]] = alpha

                elif (pattern[index] in rule.keys()) and (alpha == rule[pattern[index]]):
                    continue
                else:
                    check = False
                    break

            if check:
                res.append(word)
            rule = {}

        return res


myClass = Solution()
result = myClass.findAndReplacePattern(["abc", "deq", "mee", "aqq", "dkd", "ccc", "aaa"], "zxx")
print(result)
